Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +468,188 @@ def restore_snapshot_online(
+                  if (int(dow_t) > 6 or int(dow_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(dow_t)>1 or dow_t == ""):
+                    return (False, False, "Invalid task freq", {})
+           else:
+              return (False, False, "Invalid task freq", {})
+          else:
+              return (False, False, "Invalid task freq", {})
+          freq = task_freq_custom
+
+        if task_freq_custom and task_freq:
+            return (False, False, "Please Enter either task_freq or task_freq_custom", {})    
+        if not client_obj.scheduleExists(schedule_name):
+           cmd = ["createsv"]
+           if read_only:
+              cmd.append("-ro")
+           if expirationHours != None:
+              cmd.append("-exp")
+              cmd.append(str(expirationHours)+"h")
+           if retentionHours != None:
+                 cmd.append("-retain")
+                 cmd.append(str(retentionHours)+"h")
+           snap_string = ".@y@@m@@d@@H@@M@@S@"
+           cmd.append(snapshot_name+"-"+base_volume_name+snap_string)
+           cmd.append(base_volume_name)
+           if task_freq:
+              freq="@"+task_freq
+           cmd = ' '.join(cmd)
+           client_obj.createSchedule(
+                schedule_name, cmd, freq)
+        else:
+            return (True, False, "Schedule already Exist", {})
+    except Exception as e:
+        return (False, "False", "Schedule creation failed | %s" % (e), {})
+    finally:
+        client_obj.logout()
+    return (
+        True,
+        True,
+        "Created Schedule %s successfully." %
+       schedule_name,
+        {})
+                
+def delete_schedule(
+        client_obj,
+        storage_system_ip,
+        storage_system_username,
+        storage_system_password,
+        schedule_name):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Schedule delete failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "Schedule delete failed. Schedule name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "Schedule create failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        if client_obj.scheduleExists(schedule_name):
+           client_obj.deleteSchedule(schedule_name)
+        else:
+           return (True, False, "Schedule does not exist", {})
+    except Exception as e:
+        return (False, False, "Schedule delete failed | %s" % (e), {})
+    finally:
+        client_obj.logout()
```

Comment:
Can we do away with WSAPI login and logout and just use SSHOptions? Is that possible?